src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -0,0 +1,498 @@
+     */
+    byte[] getAttributeAsByteArray( final String attributeName );
+
+    /**
+     * Set an integer-valued attribute on the read.
+     *
+     * @param attributeName Name of the attribute to set. Must be legal according to {@link ReadUtils#assertAttributeNameIsLegal}
+     * @param attributeValue Integer value of the attribute (may be {@code null})
+     * @throws IllegalArgumentException if the attribute name is illegal according to {@link ReadUtils#assertAttributeNameIsLegal}
+     */
+    void setAttribute( final String attributeName, final Integer attributeValue );
+
+    /**
+     * Set a String-valued attribute on the read.
+     *
+     * @param attributeName Name of the attribute to set. Must be legal according to {@link ReadUtils#assertAttributeNameIsLegal}
+     * @param attributeValue String value of the attribute (may be {@code null})
+     * @throws IllegalArgumentException if the attribute name is illegal according to {@link ReadUtils#assertAttributeNameIsLegal}
+     */
+    void setAttribute( final String attributeName, final String attributeValue );
+
+    /**
+     * Set a byte array attribute on the read.
+     *
+     * @param attributeName Name of the attribute to set. Must be legal according to {@link ReadUtils#assertAttributeNameIsLegal}
+     * @param attributeValue byte array value of the attribute (may be {@code null} or empty)
+     * @throws IllegalArgumentException if the attribute name is illegal according to {@link ReadUtils#assertAttributeNameIsLegal}
+     */
+    void setAttribute( final String attributeName, final byte[] attributeValue );
+
+    /**
+     * Clear an individual attribute on the read.
+     *
+     * @param attributeName Name of the attribute to clear. Must be legal according to {@link ReadUtils#assertAttributeNameIsLegal}
+     * @throws IllegalArgumentException if the attribute name is illegal according to {@link ReadUtils#assertAttributeNameIsLegal}
+     */
+    void clearAttribute( final String attributeName );
+
+    /**
+     * Clear all attributes on the read.
+     */
+    void clearAttributes();
+
+    /**
+     * @return A copy of this read. The copy will not necessarily be a 100% independent deep copy, but should
+     *         be safe to use provided that callers do not directly modify mutable values returned from the
+     *         get methods.
+     *
+     * TODO: guarantee a deep copy here (requires writing a deep copy method for SAMRecord; copy() for
+     * TODO: Google Genomics reads already performs a deep copy)
+     */
+    GATKRead copy();
+
+    /**
+     * Convert this read into a SAMRecord.
+     *
+     * Warning: the return value is not guaranteed to be independent from this read (eg., if the read
+     * is already in SAMRecord format, no copy will be made).
+     *
+     * @param header required header for the SAMRecord
+     * @return This read as a SAMRecord
+     */
+    SAMRecord convertToSAMRecord( final SAMFileHeader header );
```

Comment:
They need direct access to the encapsulated read in order to avoid unnecessary conversion operations.